akka-remote/src/main/scala/akka/remote/artery/tcp/SSLEngineProvider.scala

```diff
@@ -84,16 +85,33 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
   val SSLRequireMutualAuthentication: Boolean = config.getBoolean("require-mutual-authentication")
   val HostnameVerification: Boolean = config.getBoolean("hostname-verification")
 
-  private lazy val sslContext: SSLContext = {
-    // log hostname verification warning once
-    if (HostnameVerification)
-      log.debug("TLS/SSL hostname verification is enabled.")
-    else
-      log.info(
-        LogMarker.Security,
-        "TLS/SSL hostname verification is disabled. See Akka reference documentation for more information.")
+  private val SSLContextCacheTime: FiniteDuration = {
+    import akka.util.JavaDurationConverters._
+    config.getDuration("ssl-context-cache-ttl").asScala
+  }
+
+  // log hostname verification warning once
+  if (HostnameVerification)
+    log.debug("TLS/SSL hostname verification is enabled.")
+  else
+    log.info(
+      LogMarker.Security,
+      "TLS/SSL hostname verification is disabled. See Akka reference documentation for more information.")
 
-    constructContext()
+  private def sslContext: SSLContext = getFromCache
+
+  @volatile private var contextRef : Option[CachedSSLContext] = None
```

Comment:
There's no need for this to be `@volatile` if `getFromCache` is `synchronized`. But if `getFromCache` is not `synchronized` (as I argue it shouldn't be), then this `@volatile` is needed.

Reply:
rename to `cachedContext`